File: 3243/3243.cs
public class Solution {
    public int ShortestDistanceAfterRoadAddit(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(ShortestDistanceAfterRoadAddit(root.left), ShortestDistanceAfterRoadAddit(root.right));
    }
}

File: 3243/3243.cpp
class Solution {
public:
    int shortestDistanceAfterRoadAddit(TreeNode* root) {
        if(!root) return 0;
        return 1+max(shortestDistanceAfterRoadAddit(root->left),shortestDistanceAfterRoadAddit(root->right));
    }
};